### DISCUSSION TITLE
How to call a function as callback in addField?

### DISCUSSION THREAD

**Discussion by cod3rshotout** [Laravel-Backpack/community-forum#418] [Q&A (Help)] [answered]
I'm using spatie for manage the permissions, they are stored in the following way:

<img src="https://i.imgur.com/1172JRf.jpg">

So I'm using this addField properties to display it:

    'label' => ucfirst(trans('backpack::permissionmanager.permission_singular')),
    'name' => 'permissions',
    'entity' => 'permissions',
    'entity_primary' => 'roles',
    'attribute' => 'name',
    'model' => config('permission.models.permission'),
    'pivot' => true,
    'number_columns' => 3,

what I'm trying to do is load a translation of each permission from my custom language file, but I cannot find any method that allow me to do this.
Currently I have this visualization:

<img src="https://i.imgur.com/FSvht6h.jpg">

**Comment by o15a3d4l11s2** [comment, marked as answer]
The first idea that comes to my mind is to use an [accessor](https://laravel.com/docs/10.x/eloquent-mutators) together with the `attribute` property.
You can do something like this instead of the current `name`
```
'attribute' => 'name_translated'
```
and then in your model add the accessor like this:
Laravel 9:
```
protected function nameTranslated(): Attribute
{
    return Attribute::make(
        get: fn ($value) => __($this->name),
    );
}
```
Laravel 8:
```
public function getNameTranslatedAttribute($value)
{
    return __($this->name);
}
```
Most probably you will have to customise the translation function `__($value)` to your needs.
If you think that in all places where you read the `name` you would like it translated, then you could just make the accessor for `name` and not for `nameTranslated`.

**Reply by cod3rshotout** [reply]
@o15a3d4l11s2 is possible do the same using addColumn instead of addField? Because I tried and actually your solution works just for addField

**Reply by o15a3d4l11s2** [reply]
Usually for columns you can also do the same (use `attribute`). What is the type of your column? Since it is related to users and roles I assume it is `select_mutiple` with pivot? Is there a specific error that you get or any more details that describe what happens exactly when you use the same `attribute`?

**Reply by cod3rshotout** [reply]
@o15a3d4l11s2 is another controller which use the same model "Permission", in the List operation I'm trying to display the translated name, this is the column declaration: ` $this->crud->addColumn([
            'name'  => 'name',
            //'label' => trans('backpack::permissionmanager.name'),
            'type'  => 'text',
            'attribute' => 'name_translated'
        ]);`

**Reply by o15a3d4l11s2** [reply]
Sorry for the late response, hope you solved it, but in any case, here is my feedback: columns work in slightly different way, so you can just add the column with `name` => `name_translated`, remove the `attribute` and it should work.
